fsspec/implementations/ftp.py

```diff
@@ -106,7 +105,7 @@ def ls(self, path, detail=True):
     def info(self, path, **kwargs):
         # implement with direct method
         path = self._strip_protocol(path)
-        files = self.ls(self._parent(path), True)
+        files = [x for x in self.ls(self._parent(path), True) if x['type'] == 'file']
```

Comment:
So, I believe you should be able to get an `info()` of a directory (should return a dict where `type=="dir"`). What happens if instead of this, in L86 we exclude those pdir and cdir entries only?

Reply:
Yes- that's better I think. I can't think of any case where cdir/pdir would be useful.